include/posix/pthread.h

```diff
@@ -383,4 +393,151 @@ int pthread_barrierattr_getpshared(const pthread_barrierattr_t *, int *);
 int pthread_barrierattr_setpshared(pthread_barrierattr_t *, int);
 */
 
+/* Base Pthread related APIs */
+
+/**
+ * @brief get pid of current execueting thread.
+ *
+ * See IEEE 1003.1
+ */
+static inline pthread_t pthread_self(void)
+{
+	return (pthread_t)k_current_get();
+}
+
+/**
+ * @brief POSIX pthread attributes destroy
+ *
+ * See IEEE 1003.1
+ */
+static inline int pthread_attr_destroy(pthread_attr_t *attr)
+{
+	return 0;
+}
+
+/**
+ * @brief POSIX get pthread scheduling parameter
+ *
+ * See IEEE 1003.1
+ */
+static inline int pthread_attr_getschedparam(const pthread_attr_t *attr,
+					     struct sched_param *schedparam)
+{
+	schedparam->priority = attr->priority;
+	return 0;
+}
+
+/**
+ * @brief POSIX get pthread stack inforamtion from thread attribute
+ *
+ * See IEEE 1003.1
+ */
+static inline int pthread_attr_getstack(const pthread_attr_t *attr,
+					void **stackaddr, size_t *stacksize)
+{
+	*stackaddr = attr->stack;
+	*stacksize = attr->stacksize;
+	return 0;
+}
+
+/**
+ * @brief POSIX get stack size from thread attribute
+ *
+ * See IEEE 1003.1
+ */
+static inline int pthread_attr_getstacksize(const pthread_attr_t *attr,
+					    size_t *stacksize)
+{
+	*stacksize = attr->stacksize;
+	return 0;
+}
+/**
+ * @brief are Both pthreads same
+ *
+ * See IEEE 1003.1
+ */
+static inline int pthread_equal(pthread_t pt1, pthread_t pt2)
+{
+	return (pt1 == pt2);
+}
+
+/**
+ * @brief get pid of current execueting thread.
```

Comment:
Typo: `execueting`

Also, this is a function to set the schedpolicy for a Pthread attribute object.

Reply:
Addressed